src/copy.rs

```diff
@@ -0,0 +1,89 @@
+use std::fmt;
+
+use serde::{de, ser};
+
+/// Defines a supported clipboard copy command
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CopyCmd {
+    /// Command to perform a copy to the clipboard
+    pub(crate) cmd: &'static str,
+    /// Arguments to pass to the copy command
+    pub(crate) args: Vec<&'static str>,
+}
+
+/// Defines the supported clipboard copy command.
+/// A `Vec<CopyCmd>` is returned according to the conditional compilation on the detected OS.
+/// The first item declared is considered the default copy command.
+fn get_supported_copy_cmds() -> Vec<CopyCmd> {
+    #[cfg(target_os = "linux")]
+    let cmds = vec![
+        CopyCmd {
+            cmd: "xclip",
+            args: vec!["-in", "-selection", "clipboard"],
+        },
+        CopyCmd {
+            cmd: "xsel",
+            args: vec!["-ib"],
+        },
+    ];
+    #[cfg(target_os = "macos")]
+    let cmds = vec![CopyCmd {
+        cmd: "pbcopy",
+        args: vec![],
+    }];
+    #[cfg(target_os = "android")]
+    let cmds = vec![CopyCmd {
+        cmd: "termux-clipboard-set",
+        args: vec![],
+    }];
+    cmds
+}
+
+/// Returns the default `CopyCmd` according to the conditional compilation on the detected OS
+/// which is the first item of the `Vec<CopyCmd>` obtained by `get_supported_copy_cmds()`.
+pub(crate) fn get_default_copy_cmd() -> CopyCmd {
+    let copy_cmds = get_supported_copy_cmds();
+    // We always have at least one element in this array
+    copy_cmds[0].to_owned()
```

Comment:
Instead, could make this return a Result and give a message like "No default command found for detected OS. Please add a supported command to your configuration file (as copy_cmd)"